### PR TITLE
8267262: com/sun/net/httpserver/Filter improve API documentation of static methods

### PR DESCRIPTION
**Pull request by FrauBoes**
The filter operation `Consumer` that is passed to the factory methods can throw an unchecked exception. This change adds a note on the exception handling in that case. It also adds a clarification to `Filter::afterHandler` on the relation of the filter operation and the client receiving the response.

CSR: https://bugs.openjdk.java.net/browse/JDK-8267379

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8267262](https://bugs.openjdk.java.net/browse/JDK-8267262): com/sun/net/httpserver/Filter improve API documentation of static methods

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**) ⚠️ Review applies to ef8ffbfe0db9c732da7cf8eac05bdc7461efaf81
 * [Chris Hegarty](https://openjdk.java.net/census#chegar) (@ChrisHegarty - **Reviewer**) ⚠️ Review applies to ef8ffbfe0db9c732da7cf8eac05bdc7461efaf81

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4085/head:pull/4085` \
`$ git checkout pull/4085`

Update a local copy of the PR: \
`$ git checkout pull/4085` \
`$ git pull https://git.openjdk.java.net/jdk pull/4085/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4085`

View PR using the GUI difftool: \
`$ git pr show -t 4085`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4085.diff">https://git.openjdk.java.net/jdk/pull/4085.diff</a>

</details>

### CODE REVIEW

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java [hunk 1]
```diff
@@ -207,7 +209,8 @@ public String description() {
      * executed. The filter {@code operation} is not expected to handle the
      * exchange or {@linkplain HttpExchange#sendResponseHeaders(int, long) send the response headers}.
      * Doing so is likely to fail, since the exchange has commonly been handled
-     * before the operation is invoked.
+     * before the {@code operation} is invoked. Correspondingly, the client
+     * commonly receives the response before the filter operation is executed.
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java:213]: I'd rather replace `commonly` with `may`: this is essentially racy and you cannot assume that any of the outcome is more likely than the other.

```
     * [...]  Correspondingly, the client
     * may receive the response before the filter operation is executed.

Comment (ChrisHegarty) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java:213]: Maybe rephrase? Who knows what the client will received, if anything - since the bytes will be sent over the network.

Maybe something like:
  More specifically, the response will typically be sent before the filter operation is executed.

**Review by dfuch — APPROVED** [review]

**Review by ChrisHegarty — APPROVED** [review]

**Comment by dfuch** [comment]
I suspect this need a CSR...
/csr needed

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java [hunk 2]
```diff
@@ -207,7 +209,8 @@ public String description() {
      * executed. The filter {@code operation} is not expected to handle the
      * exchange or {@linkplain HttpExchange#sendResponseHeaders(int, long) send the response headers}.
      * Doing so is likely to fail, since the exchange has commonly been handled
-     * before the operation is invoked.
+     * before the {@code operation} is invoked. More specifically, the response
+     * will typically be sent before the filter {@code operation} is executed.
      *
```
Comment (Michael-Mc-Mahon) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java:214]: I think I'd prefer "may be" rather than "will typically be" here as well. Best not to make any claim about what is more likely to happen.

**Review by Michael-Mc-Mahon — CHANGES_REQUESTED** [review]

**Comment by FrauBoes** [comment]
CSR for review: https://bugs.openjdk.java.net/browse/JDK-8267379
